sdk/storage/azure-storage-blob/azure/storage/blob/_shared/models.py

```diff
@@ -249,32 +239,26 @@ class ResourceTypes(object):
         A string representing the resource types.
     """
 
-    SERVICE = None  # type: ResourceTypes
-    CONTAINER = None  # type: ResourceTypes
-    OBJECT = None  # type: ResourceTypes
-
-    def __init__(self, service=False, container=False, object=False, _str=None):  # pylint: disable=redefined-builtin
-        if not _str:
-            _str = ''
-        self.service = service or ('s' in _str)
-        self.container = container or ('c' in _str)
-        self.object = object or ('o' in _str)
-
-    def __or__(self, other):
-        return ResourceTypes(_str=str(self) + str(other))
-
-    def __add__(self, other):
-        return ResourceTypes(_str=str(self) + str(other))
-
-    def __str__(self):
-        return (('s' if self.service else '') +
+    def __init__(self, service=False, container=False, object=False):  # pylint: disable=redefined-builtin
+        self.service = service
+        self.container = container
+        self.object = object
+        self._str = (('s' if self.service else '') +
                 ('c' if self.container else '') +
                 ('o' if self.object else ''))
 
+    def __str__(self):
+        return self._str
+
+    @classmethod
+    def from_string(cls, string):
+        service = 's' in string
+        container = 'c' in string
+        object = 'o' in string
```

Comment:
object is a reserved keyword in python...we dont want to redefine it...perhaps rename these to res_service, res_object etc?